src/js/controller/settings/exportimage/ImageExportController.js

```diff
@@ -7,7 +7,15 @@
     this.gifExportController = new ns.GifExportController(piskelController);
   };
 
+  pskl.utils.inherit(ns.ImageExportController, pskl.controller.settings.AbstractSettingController);
+
   ns.ImageExportController.prototype.init = function () {
+    // Output Scaling Factor
+    var scalingFactorInput = document.querySelector('.scaling-factor-input');
+    scalingFactorInput.value = pskl.UserSettings.get(pskl.UserSettings.EXPORT_SCALING);
+    this.addEventListener(scalingFactorInput, 'change', this.onScalingFactorChange_);
```

Comment:
We need to make sure the event listeners added are removed when the panel is closed. This is handled by  pskl.controller.settings.AbstractSettingController#destroy . The PngExportController and GifExportController both extend AbstractSettingController without overriding the destroy method. All their events are automatically cleaned up.

Here you can either add a call to ```this.superclass.destroy.call(this);``` at the end of the destroy method. Or you can move the event's logic to the PngExportController.